# This file is part of HiGFXback

REQUIRES="autotools-wrappers-ac-build expat-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69)
autoconf; autoheader
./configure --enable-shared --with-system-expat --prefix=/

# build
make

# install
install -d $DESTDIR/bin
install python $DESTDIR/bin/python2
install -d $DESTDIR/include/python2
install -m 644 pyconfig.h $DESTDIR/include/python2
install -m 644 Include/abstract.h $DESTDIR/include/python2
install -m 644 Include/boolobject.h $DESTDIR/include/python2
install -m 644 Include/bufferobject.h $DESTDIR/include/python2
install -m 644 Include/bytearrayobject.h $DESTDIR/include/python2
install -m 644 Include/bytesobject.h $DESTDIR/include/python2
install -m 644 Include/cellobject.h $DESTDIR/include/python2
install -m 644 Include/ceval.h $DESTDIR/include/python2
install -m 644 Include/classobject.h $DESTDIR/include/python2
install -m 644 Include/cobject.h $DESTDIR/include/python2
install -m 644 Include/code.h $DESTDIR/include/python2
install -m 644 Include/codecs.h $DESTDIR/include/python2
install -m 644 Include/compile.h $DESTDIR/include/python2
install -m 644 Include/complexobject.h $DESTDIR/include/python2
install -m 644 Include/descrobject.h $DESTDIR/include/python2
install -m 644 Include/dictobject.h $DESTDIR/include/python2
install -m 644 Include/dtoa.h $DESTDIR/include/python2
install -m 644 Include/enumobject.h $DESTDIR/include/python2
install -m 644 Include/eval.h $DESTDIR/include/python2
install -m 644 Include/fileobject.h $DESTDIR/include/python2
install -m 644 Include/floatobject.h $DESTDIR/include/python2
install -m 644 Include/funcobject.h $DESTDIR/include/python2
install -m 644 Include/genobject.h $DESTDIR/include/python2
install -m 644 Include/import.h $DESTDIR/include/python2
install -m 644 Include/intobject.h $DESTDIR/include/python2
install -m 644 Include/intrcheck.h $DESTDIR/include/python2
install -m 644 Include/iterobject.h $DESTDIR/include/python2
install -m 644 Include/listobject.h $DESTDIR/include/python2
install -m 644 Include/longobject.h $DESTDIR/include/python2
install -m 644 Include/memoryobject.h $DESTDIR/include/python2
install -m 644 Include/methodobject.h $DESTDIR/include/python2
install -m 644 Include/modsupport.h $DESTDIR/include/python2
install -m 644 Include/moduleobject.h $DESTDIR/include/python2
install -m 644 Include/object.h $DESTDIR/include/python2
install -m 644 Include/objimpl.h $DESTDIR/include/python2
install -m 644 Include/patchlevel.h $DESTDIR/include/python2
install -m 644 Include/pyarena.h $DESTDIR/include/python2
install -m 644 Include/pycapsule.h $DESTDIR/include/python2
install -m 644 Include/pyctype.h $DESTDIR/include/python2
install -m 644 Include/pydebug.h $DESTDIR/include/python2
install -m 644 Include/pyerrors.h $DESTDIR/include/python2
install -m 644 Include/pyfpe.h $DESTDIR/include/python2
install -m 644 Include/pymacconfig.h $DESTDIR/include/python2
install -m 644 Include/pymath.h $DESTDIR/include/python2
install -m 644 Include/pymem.h $DESTDIR/include/python2
install -m 644 Include/pyport.h $DESTDIR/include/python2
install -m 644 Include/pystate.h $DESTDIR/include/python2
install -m 644 Include/pystrcmp.h $DESTDIR/include/python2
install -m 644 Include/pystrtod.h $DESTDIR/include/python2
install -m 644 Include/Python.h $DESTDIR/include/python2
install -m 644 Include/pythonrun.h $DESTDIR/include/python2
install -m 644 Include/rangeobject.h $DESTDIR/include/python2
install -m 644 Include/setobject.h $DESTDIR/include/python2
install -m 644 Include/sliceobject.h $DESTDIR/include/python2
install -m 644 Include/stringobject.h $DESTDIR/include/python2
install -m 644 Include/sysmodule.h $DESTDIR/include/python2
install -m 644 Include/traceback.h $DESTDIR/include/python2
install -m 644 Include/tupleobject.h $DESTDIR/include/python2
install -m 644 Include/unicodeobject.h $DESTDIR/include/python2
install -m 644 Include/warnings.h $DESTDIR/include/python2
install -m 644 Include/weakrefobject.h $DESTDIR/include/python2
ln -sf python2 $DESTDIR/include/python2.7
install -d $DESTDIR/lib
install libpython2.7.so.1 $DESTDIR/lib
ln -sf libpython2.7.so.1 $DESTDIR/lib/libpython2.7.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 Misc/python.pc $DESTDIR/lib/pkgconfig/python2.pc
install -d $DESTDIR/lib/python2
install -m 644 `cat pybuilddir.txt`/_sysconfigdata.py $DESTDIR/lib/python2
install -m 644 Lib/__future__.py $DESTDIR/lib/python2
install -m 644 Lib/_weakrefset.py $DESTDIR/lib/python2
install -m 644 Lib/_abcoll.py $DESTDIR/lib/python2
install -m 644 Lib/abc.py $DESTDIR/lib/python2
install -m 644 Lib/argparse.py $DESTDIR/lib/python2
install -m 644 Lib/base64.py $DESTDIR/lib/python2
install -m 644 Lib/codecs.py $DESTDIR/lib/python2
install -m 644 Lib/collections.py $DESTDIR/lib/python2
install -m 644 Lib/copy.py $DESTDIR/lib/python2
install -m 644 Lib/copy_reg.py $DESTDIR/lib/python2
install -m 644 Lib/fnmatch.py $DESTDIR/lib/python2
install -m 644 Lib/functools.py $DESTDIR/lib/python2
install -m 644 Lib/genericpath.py $DESTDIR/lib/python2
install -m 644 Lib/getopt.py $DESTDIR/lib/python2
install -m 644 Lib/gettext.py $DESTDIR/lib/python2
install -m 644 Lib/glob.py $DESTDIR/lib/python2
install -m 644 Lib/heapq.py $DESTDIR/lib/python2
install -m 644 Lib/io.py $DESTDIR/lib/python2
install -m 644 Lib/keyword.py $DESTDIR/lib/python2
install -m 644 Lib/linecache.py $DESTDIR/lib/python2
install -m 644 Lib/locale.py $DESTDIR/lib/python2
install -m 644 Lib/optparse.py $DESTDIR/lib/python2
install -m 644 Lib/os.py $DESTDIR/lib/python2
install -m 644 Lib/pickle.py $DESTDIR/lib/python2
install -m 644 Lib/posixpath.py $DESTDIR/lib/python2
install -m 644 Lib/py_compile.py $DESTDIR/lib/python2
install -m 644 Lib/re.py $DESTDIR/lib/python2
install -m 644 Lib/site.py $DESTDIR/lib/python2
install -m 644 Lib/socket.py $DESTDIR/lib/python2
install -m 644 Lib/sre_compile.py $DESTDIR/lib/python2
install -m 644 Lib/sre_constants.py $DESTDIR/lib/python2
install -m 644 Lib/sre_parse.py $DESTDIR/lib/python2
install -m 644 Lib/stat.py $DESTDIR/lib/python2
install -m 644 Lib/string.py $DESTDIR/lib/python2
install -m 644 Lib/StringIO.py $DESTDIR/lib/python2
install -m 644 Lib/struct.py $DESTDIR/lib/python2
install -m 644 Lib/subprocess.py $DESTDIR/lib/python2
install -m 644 Lib/sysconfig.py $DESTDIR/lib/python2
install -m 644 Lib/textwrap.py $DESTDIR/lib/python2
install -m 644 Lib/traceback.py $DESTDIR/lib/python2
install -m 644 Lib/types.py $DESTDIR/lib/python2
install -m 644 Lib/urllib.py $DESTDIR/lib/python2
install -m 644 Lib/urlparse.py $DESTDIR/lib/python2
install -m 644 Lib/UserDict.py $DESTDIR/lib/python2
install -m 644 Lib/warnings.py $DESTDIR/lib/python2
install -m 644 Lib/weakref.py $DESTDIR/lib/python2
install -d $DESTDIR/lib/python2/encodings
install -m 644 Lib/encodings/__init__.py $DESTDIR/lib/python2/encodings
install -m 644 Lib/encodings/aliases.py $DESTDIR/lib/python2/encodings
install -m 644 Lib/encodings/ascii.py $DESTDIR/lib/python2/encodings
install -d $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/_collections.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/_elementtree.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/_functools.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/_io.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/_socket.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/_struct.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/binascii.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/fcntl.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/itertools.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/math.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/operator.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/pyexpat.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/select.so $DESTDIR/lib/python2/lib-dynload
install `cat pybuilddir.txt`/time.so $DESTDIR/lib/python2/lib-dynload
install -d $DESTDIR/lib/python2/xml
install -m 644 Lib/xml/__init__.py $DESTDIR/lib/python2/xml
install -d $DESTDIR/lib/python2/xml/etree
install -m 644 Lib/xml/etree/__init__.py $DESTDIR/lib/python2/xml/etree
install -m 644 Lib/xml/etree/cElementTree.py $DESTDIR/lib/python2/xml/etree
install -m 644 Lib/xml/etree/ElementTree.py $DESTDIR/lib/python2/xml/etree
install -d $DESTDIR/lib/python2/xml/parsers
install -m 644 Lib/xml/parsers/__init__.py $DESTDIR/lib/python2/xml/parsers
install -m 644 Lib/xml/parsers/expat.py $DESTDIR/lib/python2/xml/parsers
install -d $DESTDIR/lib/python2/xml/sax
install -m 644 Lib/xml/sax/__init__.py $DESTDIR/lib/python2/xml/sax
install -m 644 Lib/xml/sax/_exceptions.py $DESTDIR/lib/python2/xml/sax
install -m 644 Lib/xml/sax/expatreader.py $DESTDIR/lib/python2/xml/sax
install -m 644 Lib/xml/sax/handler.py $DESTDIR/lib/python2/xml/sax
install -m 644 Lib/xml/sax/saxutils.py $DESTDIR/lib/python2/xml/sax
install -m 644 Lib/xml/sax/xmlreader.py $DESTDIR/lib/python2/xml/sax
test -L $DESTDIR/lib/python2.7 || ln -s python2 $DESTDIR/lib/python2.7

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/python2-build.pc << EOF
Name: Python
Version: 2.7.18
Description: Python programming language
Requires: $REQUIRES

devel=\\
/include/python2/abstract.h \\
/include/python2/boolobject.h \\
/include/python2/bufferobject.h \\
/include/python2/bytearrayobject.h \\
/include/python2/bytesobject.h \\
/include/python2/cellobject.h \\
/include/python2/ceval.h \\
/include/python2/classobject.h \\
/include/python2/cobject.h \\
/include/python2/code.h \\
/include/python2/codecs.h \\
/include/python2/compile.h \\
/include/python2/complexobject.h \\
/include/python2/descrobject.h \\
/include/python2/dictobject.h \\
/include/python2/dtoa.h \\
/include/python2/enumobject.h \\
/include/python2/eval.h \\
/include/python2/fileobject.h \\
/include/python2/floatobject.h \\
/include/python2/funcobject.h \\
/include/python2/genobject.h \\
/include/python2/import.h \\
/include/python2/intobject.h \\
/include/python2/intrcheck.h \\
/include/python2/iterobject.h \\
/include/python2/listobject.h \\
/include/python2/longobject.h \\
/include/python2/memoryobject.h \\
/include/python2/methodobject.h \\
/include/python2/modsupport.h \\
/include/python2/moduleobject.h \\
/include/python2/object.h \\
/include/python2/objimpl.h \\
/include/python2/patchlevel.h \\
/include/python2/pyarena.h \\
/include/python2/pycapsule.h \\
/include/python2/pyconfig.h \\
/include/python2/pyctype.h \\
/include/python2/pydebug.h \\
/include/python2/pyerrors.h \\
/include/python2/pyfpe.h \\
/include/python2/pymacconfig.h \\
/include/python2/pymath.h \\
/include/python2/pymem.h \\
/include/python2/pyport.h \\
/include/python2/pystate.h \\
/include/python2/pystrcmp.h \\
/include/python2/pystrtod.h \\
/include/python2/Python.h \\
/include/python2/pythonrun.h \\
/include/python2/rangeobject.h \\
/include/python2/setobject.h \\
/include/python2/sliceobject.h \\
/include/python2/stringobject.h \\
/include/python2/sysmodule.h \\
/include/python2/traceback.h \\
/include/python2/tupleobject.h \\
/include/python2/unicodeobject.h \\
/include/python2/warnings.h \\
/include/python2/weakrefobject.h \\
/include/python2.7 \\
/lib/libpython2.7.so \\
/lib/pkgconfig/python2.pc

exec=\\
/bin/python2 \\
/lib/libpython2.7.so.1 \\
/lib/python2/__future__.py \\
/lib/python2/_abcoll.py \\
/lib/python2/_sysconfigdata.py \\
/lib/python2/_weakrefset.py \\
/lib/python2/abc.py \\
/lib/python2/argparse.py \\
/lib/python2/base64.py \\
/lib/python2/codecs.py \\
/lib/python2/collections.py \\
/lib/python2/copy.py \\
/lib/python2/copy_reg.py \\
/lib/python2/fnmatch.py \\
/lib/python2/functools.py \\
/lib/python2/genericpath.py \\
/lib/python2/getopt.py \\
/lib/python2/gettext.py \\
/lib/python2/glob.py \\
/lib/python2/heapq.py \\
/lib/python2/io.py.py \\
/lib/python2/keyword.py \\
/lib/python2/linecache.py \\
/lib/python2/locale.py \\
/lib/python2/optparse.py \\
/lib/python2/os.py \\
/lib/python2/pickle.py \\
/lib/python2/posixpath.py \\
/lib/python2/py_compile.py \\
/lib/python2/re.py \\
/lib/python2/site.py \\
/lib/python2/socket.py \\
/lib/python2/sre_compile.py \\
/lib/python2/sre_constants.py \\
/lib/python2/sre_parse.py \\
/lib/python2/stat.py \\
/lib/python2/string.py \\
/lib/python2/StringIO.py \\
/lib/python2/struct.py \\
/lib/python2/subprocess.py \\
/lib/python2/sysconfig.py \\
/lib/python2/textwrap.py \\
/lib/python2/traceback.py \\
/lib/python2/types.py \\
/lib/python2/urllib.py \\
/lib/python2/urlparse.py \\
/lib/python2/UserDict.py \\
/lib/python2/warnings.py \\
/lib/python2/weakref.py \\
/lib/python2/encodings/__init__.py \\
/lib/python2/encodings/aliases.py \\
/lib/python2/encodings/ascii.py \\
/lib/python2/lib-dynload/_collections.so \\
/lib/python2/lib-dynload/_elementtree.so \\
/lib/python2/lib-dynload/_functools.so \\
/lib/python2/lib-dynload/_io.so \\
/lib/python2/lib-dynload/_socket.so \\
/lib/python2/lib-dynload/_struct.so \\
/lib/python2/lib-dynload/binascii.so \\
/lib/python2/lib-dynload/fcntl.so \\
/lib/python2/lib-dynload/itertools.so \\
/lib/python2/lib-dynload/math.so \\
/lib/python2/lib-dynload/operator.so \\
/lib/python2/lib-dynload/pyexpat.so \\
/lib/python2/lib-dynload/select.so \\
/lib/python2/lib-dynload/time.so \\
/lib/python2/xml/__init__.py \\
/lib/python2/xml/etree/__init__.py \\
/lib/python2/xml/etree/cElementTree.py \\
/lib/python2/xml/etree/ElementTree.py \\
/lib/python2/xml/parsers/__init__.py \\
/lib/python2/xml/parsers/expat.py \\
/lib/python2/xml/sax/__init__.py \\
/lib/python2/xml/sax/_exceptions.py \\
/lib/python2/xml/sax/expatreader.py \\
/lib/python2/xml/sax/handler.py \\
/lib/python2/xml/sax/saxutils.py \\
/lib/python2/xml/sax/xmlreader.py \\
/lib/python2.7
EOF

exit
--- Python-2.7.18.orig/configure.ac
+++ Python-2.7.18/configure.ac
@@ -79,7 +79,7 @@
 VERSION=PYTHON_VERSION
 
 AC_SUBST(SOVERSION)
-SOVERSION=1.0
+SOVERSION=1
 
 # The later defininition of _XOPEN_SOURCE disables certain features
 # on Linux, so we need _GNU_SOURCE to re-enable them (makedev, tm_zone).
